Adds learner profiles. Closes #86 (#95)
carpentries-incubator/docker-introduction: index.md

---
This session aims to introduce the use of Docker containers with the goal of using them to effect reproducible computational environments. Such environments are useful for ensuring reproducible research outputs, for example.

After completing this session you should:
- Have an understanding of what Docker containers are, why they are useful
  and the common terminology used

> - Have a working Docker installation on your local system to allow you to
>   use Docker

Reviewer: ```suggestion
- Have a working Docker installation on your local system to allow you to
  use containers
```

- Understand how to use existing Docker containers for common tasks

> - Be able to build your own Docker containers and understand the role/syntax
>   of the Dockerfile in building containers

Reviewer: ```suggestion
- Be able to build your own Docker containers by understanding both the role
  of a `Dockerfile` in building containers, and the syntax used in `Dockerfile`s
```

> If you are looking for a lesson on using *singularity* containers (instead of Docker), see this lesson:
> * [Reproducible Computational Environments Using Containers: Introduction to Singularity](https://carpentries-incubator.github.io/singularity-introduction/)

## Learner profiles

To help give an idea of the target audience for this lesson we have included 
some example learner profiles.

> *Nelson is a graduate student in microbiology.* They have experience in running unix shell

Reviewer: ```suggestion
*Nelson is a graduate student in microbiology.* They have experience in running Unix shell
```

commands and using libraries in R for the bioinformatics workflows they have devleoped.
They are expanding their analysis to run on 3000 genomes in 200 samples and they have
started to use the local cluster to run their workflows.  The local reserach computaing
facilitator has advised them that Docker could be useful for running their workflows on
the cluster. They'd like to make use of existing containers that other bioinformaticians
have made so they want to learn how to use Docker. They would also be interested in
creating their own Docker images for other lab members and collaborators to re-use their
workflow.

*Caitlin is a second year undergraduate in computer science examining Docker for the first
time.* She has heard about Docker but does not really know what it achieves or why it is
useful. She is reasonably confident in using the Unix shell, having used it briefly in

> her first year papers. She is keen to find jump-off points to learn more about technical

Reviewer 2: If papers is considered a widely understood term, then happy to leave this, otherwise I'd suggest maybe changing this to "courses" or "modules".
```suggestion
her first year modules. She is keen to find jump-off points to learn more about technical
```

Reviewer 3: I'm pretty indifferent about this change.

details and alternative technologies that are also popular, having heard that container
technologies are widely used within industry.

> *Xu, a materials science researcher, wants to package their software for release with

Reviewer 2: ```suggestion
*Xu, a materials science researcher, wants to package her software for release with
```

a paper to help ensure reproducibility.* She has written some code that makes use of a

> series of Python libraries to undertake analysis of a compound. He wants to (or is

Reviewer 2: ```suggestion
series of Python libraries to undertake analysis of a compound. She wants to (or is
```

required to) make her software available as part of the paper submission. She
understands why Docker is important in helping to ensure reproducibility but not the
process and low-level detail of preparing a container and archiving it to obtain a DOI
for inclusion with the paper submission.

> *Bronwyn is a PhD student running Python/R scripts on their local laptop/workstation.*

Reviewer 4: also mix of pronouns in this one between the summary sentence here and rest of paragraph

> She is having difficulty getting all the tools they need to work because of conflicing

Reviewer: ```suggestion
She is having difficulty getting all the tools she needs to work because of conflicting
```

depenencies and little experience with package managers. She is also keen to reduce
the overhead of managing software so she can get on with her thesis research. She has
heard that Docker might be able to help out but is not confident to start exploring
this on her own and does not have access to any expertise in this within her local
research group. She currently wants to know how to use preexisting Docker containers
but may need to create her own containers in the future.

> *Virat is a grad student who is running an obscure bioinformatics tool (from a GitHub

Reviewer 4: again mix of they and he pronouns

> repo) that depends on a bunch of pre-installed other tools.* He wants to be able to

Reviewer 3: instead of "... bunch of other pre-installed other tools." -> "... bunch of other pre-installed tools."

run on multiple resources and have his undergrad assistant use the same tools. James
has command line experience and has blundered their way through complex installations

> but no formal CS background - he only knows to use containers because an departmental

Reviewer 2: ```suggestion
but no formal CS background - he only knows to use containers because a departmental
```

> repo) that depends on a bunch of pre-installed other tools.* He wants to be able to
> run on multiple resources and have his undergrad assistant use the same tools. James
> has command line experience and has blundered their way through complex installations
> but no formal CS background - he only knows to use containers because an departmental
> IT person suggested it. He is usually working from a Windows computer. He needs to
> understand how to create their own container, use it locally, and train his student
> to use it as well. 

Reviewer 2: I was quite confused by this, I wasn't clear if "James" was the undergrad but if so, that meant that the rest of the text didn't seem to read correctly. Suggesting a few edits to address this:
```suggestion
repo) that depends on a number of other tools that need to be pre-installed .* He wants to be able to
run on multiple resources and have his undergrad assistant use the same tools. Virat
has command line experience and has struggled his way through complex installations
but he has no formal CS background - he only knows to use containers because a departmental
IT person suggested it. He is usually working from a Windows computer. He needs to
understand how to create his own container, use it locally, and train his student
to use it as well. 
```
